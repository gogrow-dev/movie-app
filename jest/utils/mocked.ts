import { Category, Movie } from '../../src/store/actions/moviesActions';

export const mockedCategories: Category[] = [
  {
    id: 1,
    name: 'Category 1',
  },
  {
    id: 2,
    name: 'Category 2',
  },
];

export const mockedMovies: Movie[] = [
  {
    id: 1,
    title: 'Title',
    releaseDate: '2022-01-01',
    posterPath: '/path.png',
    voteAverage: 8.1,
    voteCount: 1234,
    backdropPath: '/path.png',
    genreIds: [],
    overview: 'Movie Overview',
    popularity: 10,
    bookmark: false,
  },
  {
    id: 2,
    title: 'Title 2',
    releaseDate: '2022-01-01',
    posterPath: '/path.png',
    voteAverage: 8.1,
    voteCount: 1234,
    backdropPath: '/path.png',
    genreIds: [],
    overview: 'Movie Overview',
    popularity: 10,
    bookmark: false,
  },
];
