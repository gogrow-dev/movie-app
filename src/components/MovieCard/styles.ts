import { StyleSheet } from 'react-native';
import colors from '../../styles/colors';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 120,
    width: '100%',
    marginBottom: 31,
  },
  image: {
    height: 120,
    width: 95,
    borderRadius: 16,
  },
  content: {
    justifyContent: 'space-between',
    marginLeft: 22,
  },
  heading: {
    fontFamily: 'Poppins-SemiBold',
    color: colors.white,
    fontSize: 12,
  },
  blurb: {
    fontFamily: 'Poppins-Regular',
    color: colors.white,
    fontSize: 12,
  },
});
