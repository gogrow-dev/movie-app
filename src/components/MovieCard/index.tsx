import { View, Text, Image } from 'react-native';
import React from 'react';
import { Movie } from '../../store/actions/moviesActions';
import styles from './styles';
import { IMAGE_BASE_URL } from '../../api';

interface IMovieCard {
  item: Movie;
}

const MovieCard = ({ item }: IMovieCard) => {
  return (
    <View style={styles.container}>
      <Image
        source={{ uri: `${IMAGE_BASE_URL}${item.posterPath}` }}
        style={styles.image}
        resizeMode="contain"
      />
      <View style={styles.content}>
        <View>
          <Text style={styles.heading}>Title:</Text>
          <Text style={styles.blurb}>{item.title}</Text>
        </View>
        <View>
          <Text style={styles.heading}>Release Date:</Text>
          <Text style={styles.blurb}>{item.releaseDate}</Text>
        </View>
        <View>
          <Text style={styles.heading}>Average Rating:</Text>
          <Text style={styles.blurb}>{item.voteAverage}</Text>
        </View>
      </View>
    </View>
  );
};

export default MovieCard;
