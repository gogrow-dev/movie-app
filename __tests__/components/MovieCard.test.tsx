import 'react-native';
import React from 'react';
import MovieCard from '../../src/components/MovieCard';
import renderer from 'react-test-renderer';
import { mockedMovies } from '../../jest/utils/mocked';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

describe('Testing MovieCard', () => {
  it('Create snapshot', () => {
    const tree = renderer.create(<MovieCard item={mockedMovies[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
