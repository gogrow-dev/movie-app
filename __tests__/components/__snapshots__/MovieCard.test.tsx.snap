// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MovieCard Create snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 120,
        "marginBottom": 31,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "https://image.tmdb.org/t/p/w300//path.png",
        }
      }
      style={
        Object {
          "borderRadius": 16,
          "height": 120,
          "width": 95,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "space-between",
          "paddingHorizontal": 22,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#EEEEEE",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 12,
            }
          }
        >
          Title:
        </Text>
        <Text
          style={
            Object {
              "color": "#EEEEEE",
              "fontFamily": "Poppins-Regular",
              "fontSize": 12,
            }
          }
        >
          Title
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#EEEEEE",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 12,
            }
          }
        >
          Release Date:
        </Text>
        <Text
          style={
            Object {
              "color": "#EEEEEE",
              "fontFamily": "Poppins-Regular",
              "fontSize": 12,
            }
          }
        >
          2022-01-01
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#EEEEEE",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 12,
            }
          }
        >
          Average Rating:
        </Text>
        <Text
          style={
            Object {
              "color": "#EEEEEE",
              "fontFamily": "Poppins-Regular",
              "fontSize": 12,
            }
          }
        >
          8.1
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "width": 40,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/bookmark.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "marginTop": 18,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/icons/star-solid.png",
            }
          }
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#EEEEEE",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 12,
              },
              Object {
                "color": "#4FCCA3",
              },
            ]
          }
        >
          1234
        </Text>
      </View>
    </View>
  </View>
</View>
`;
